matminer/featurizers/dos.py

```diff
@@ -101,6 +101,97 @@ def implementors(self):
+    of states (e.g. as opposed to partial dos in DOSFeaturizer), concentration
+    and temperature.
+    """
+    def __init__(self, dopings=None, eref="midgap", T=300, return_eref=False):
+        """
+        Args:
+            dopings ([float]): list of doping concentrations. Note that a
+                negative concentration is treated as electron majority carrier
+                (n-type) and positive for holes (p-type)
+            eref (str or int or float): energy alignment reference. Defaults
+                to midgap (equilibrium fermi). A fixed number can also be used.
+                str options: "midgap", "vbm", "cbm", "dos_fermi", "band_center"
+            T (float): absolute temperature in Kelvin
+            return_eref: if True, instead of aligning the fermi levels based
+                on eref, it (eref) will be explicitly returned as a feature
+        """
+        self.dopings = dopings or [-1e20, 1e20]
+        self.eref = eref or 0.0
+        self.T = T
+        self.return_eref = return_eref
+        self.BC = BandCenter()
+
+    def featurize(self, dos, structure=None, bandgap=None):
+        """
+        Args:
+            dos (pymatgen Dos, CompleteDos or FermiDos):
+            structure (pymatgen Structure):
+            bandgap (float): for example the experimentally measured band gap
+                or one that is calculated via more accurate methods than the
+                one used to generate dos. dos will be scissored to have the
+                same electronic band gap as bandgap.
+        Returns ([float]): list of features
+        """
+        if not hasattr(dos, 'structure') or dos.structure is None:
+            dos.structure = structure
+        dos = FermiDos(dos, bandgap=bandgap)
+        feats = []
+        eref = 0.0
+        for c in self.dopings:
+            fermi = dos.get_fermi(c=c, T=self.T, nstep=50)
+            if isinstance(self.eref, str):
+                if self.eref == "dos_fermi":
+                    eref = dos.efermi
+                elif self.eref in ["midgap", "vbm", "cbm"]:
+                    ecbm, evbm = dos.get_cbm_vbm()
+                    if self.eref == "midgap":
+                        eref = (evbm + ecbm)/2.0
+                    elif self.eref == "vbm":
+                        eref = evbm
+                    elif self.eref == "cbm":
+                        eref = ecbm
+                elif self.eref == "band center":
+                    eref = self.BC.featurize(dos.structure.composition)[0]
+                else:
+                    raise ValueError('Unsupported "eref": {}'.format(self.eref))
+            else:
+                eref = self.eref
+            if not self.return_eref:
+                fermi -= eref
+            feats.append(fermi)
+        if self.return_eref:
+            feats.append(eref)
+        return feats
+
+    def feature_labels(self):
+        """
+        Returns ([str]): list of names of the features generated by featurize
```

Comment:
Could you move this description of the features to the main class documentation? This change would make it more consistent with the other Featurizers.